core/metals.py

```diff
@@ -85,6 +86,34 @@ def on_pre_send_request_async(self, request_id: int, request: LspRequest) -> Non
                 if region.contains(point):
                     request.params['range'] = region_to_range(view, region)
 
+    def on_open_uri_async(self, uri: DocumentUri, callback: Callable[[str, str, str], None]) -> bool:
+        if not uri.startswith("jar:"):
+            return False
+
+        session = self.weaksession()
+        if not session:
+            return False
+
+        params = { "command": "file-decode", "arguments": [uri] }
+
+        def handle_response(response: Any) -> None:
+            if isinstance(response, Error) or 'error' in response:
+                handle_error("file-decode", response)
+            if response and 'value' in response:
+                basename = os.path.basename(response['requestedUri'])
```

Comment:
I think you should be passing whole `requestedUri` as title. It's probably useful to see the whole path as otherwise it could become ambiguous what the opened file is related to.

(note that I know nothing about metals and haven't checked how the URI looks like)

Reply:
Yea, you're right. We should display more than just the basename as the title.

The whole `requestedUri` might not be that useful. I'll see what I can change it to. 

Thanks for the suggestion.

Reply:
The `requestedUri` looks something like this: 

```
jar:file:///Users/xxxxxx/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/typelevel/cats-effect-kernel_3/3.5.4/cats-effect-kernel_3-3.5.4-sources.jar!/cats/effect/kernel/MonadCancel.scala
```

Using the example above, I'm thinking keeping just this: 

```
cats-effect-kernel_3-3.5.4-sources.jar!/cats/effect/kernel/MonadCancel.scala
```

Thoughts? 

cc: @ayoub-benali

Reply:
Your example make sense 👍  jar file name + the path the file